apps/framework-cli/src/cli/routines.rs

```diff
@@ -263,29 +263,34 @@ async fn initialize_project_state(
     let producer = redpanda::create_producer(project.redpanda_config.clone());
 
     info!("Starting schema directory crawl...");
-    let crawl_result =
-        process_schemas_in_dir(&schema_dir, project, &configured_client, route_table).await;
-
-    let _ = post_current_state_to_console(
-        project,
-        &configured_client,
-        &producer,
-        route_table.clone(),
-        project.console_config.clone(),
-    )
-    .await;
-
-    match crawl_result {
-        Ok(_) => {
-            info!("Schema directory crawl completed successfully");
-            Ok(())
-        }
-        Err(e) => {
-            debug!("Schema directory crawl failed");
-            debug!("Error: {:?}", e);
-            Err(e)
-        }
-    }
+
+    let _ = with_spinner("Processing schema file", || async {
```

Comment:
If I understand async Rust correctly:

`with_spinner` runs the provided function, which immediately returns a `Future`, which does nothing unless `.await`ed. You then discarded the `Future` with `let _ =`. This block basically does nothing I think.

The problem here is that `with_spinner` works with sync functions, but not async functions.

We need another version for async functions. Or better yet, migrate the other sync functions to async functions. For docker commands, we can use https://docs.rs/tokio/latest/tokio/process/struct.Command.html

Reply:
@phiSgr thanks for that catch!  Are you proposing that we:

- refactor all uses of process::Command to use the [tokio version](https://docs.rs/tokio/latest/tokio/process/struct.Command.html) 
- refactor [docker.rs](https://github.com/514-labs/moose/blob/main/apps/framework-cli/src/utilities/docker.rs) to use the [docker API for Rust](https://docs.rs/docker-api/latest/docker_api/) instead of converting its use of process::Command with tokio versions.

The second point above for rewriting the docker.rs module seems like a big lift for a logging cleanup ticket.